server/hotrod/src/main/java/org/infinispan/server/hotrod/configuration/HotRodServerConfigurationBuilder.java

```diff
@@ -125,6 +125,9 @@ public void validate() {
       if (proxyHost == null && host == null) {
          throw log.missingHostAddress();
       }
+      if (proxyPort < 0 && port == 0 && !startTransport) {
```

Comment:
why are you checking the `proxyPort`? I don't see why you can't have a `proxyPort` with a "ephemeral port".